src/obb_demo/views/play.cljs

```diff
@@ -20,17 +20,26 @@
 (defn- get-game-data
   "Gets the current game or creates a new one"
   [state]
-  (if-let [game (:index state)]
-    game
-    (let [game (processor/deployed-game)
-          game-data {:game game}]
-      (state/set-page-data! game-data)
-      game-data)))
+  (let [game-data (:index state)
+        game (:game game-data)]
+    (if game
+      game-data
+      (let [mode (get game-data :mode :annihilation)
+            game (processor/deployed-game {:mode mode})
+            game-data {:game game
+                       :mode mode}]
+        (state/set-page-data! game-data)
+        game-data))))
 
 (defn- restart-game
   "Generates and restarts a new game"
   []
-  (state/set-page-data! nil))
+  (state/set-page-data! {:mode :annihilation}))
```

Comment:
Since we're talking keeping the defaults in `game-progress`. Wouldn't it make sense to retrieve this from there as well?